src/dmd/func.d

```diff
@@ -3693,3 +3694,27 @@ extern (C++) final class DeleteDeclaration : FuncDeclaration
     }
 }
 
+/** Show supplemental error message for a function call with mismatched arguments.
+ * Params:
+ *  loc = line information for error message
+ *  fargs = arguments to function
+ *  tf = function called
+ *  failIndex = index of first argument mismatch */
+void showArgMismatch(Loc loc, Expressions* fargs, TypeFunction tf, size_t failIndex)
+{
+    if (failIndex < fargs.dim && failIndex < tf.parameters.dim)
+    {
+        auto arg = (*fargs)[failIndex];
+        auto par = (*tf.parameters)[failIndex];
```

Comment:
`failIndex` is the argument index, not parameter, so for typesafe variadics with more than one argument, no supplemental error will be shown. (This could be worked around perhaps). I'll send a PR to test this.